File: app/src/main/java/com/sky/chowder/ui/activity/DetailActivity.java
package com.sky.chowder.ui.activity;

import android.os.Bundle;
import android.support.design.widget.CollapsingToolbarLayout;
import android.support.design.widget.Snackbar;
import android.support.v7.widget.Toolbar;
import android.view.View;

import com.sky.base.BaseNoPActivity;
import com.sky.chowder.R;

/**
 * @author sky QQ:1136096189
 * @Description:
 * @date 15/11/28 下午2:22
 */
public class DetailActivity extends BaseNoPActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);
        Toolbar toolbar = (Toolbar) this.findViewById(R.id.toolbar);
        toolbar.setNavigationIcon(R.mipmap.ic_back);


        setSupportActionBar(toolbar);
        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
        CollapsingToolbarLayout collapsingToolbar =
                (CollapsingToolbarLayout) findViewById(R.id.collapsing_toolbar);
        collapsingToolbar.setTitle("我的课程");
    }

    @Override
    protected int getLayoutResId() {
        return R.layout.activity_detail;
    }

    public void checkin(View view) {
        Snackbar.make(view, "checkin success!", Snackbar.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/sky/chowder/ui/activity/DetailActivity.kt
package com.sky.chowder.ui.activity

import android.support.design.widget.Snackbar
import android.view.View
import com.sky.base.BaseNoPActivity
import com.sky.chowder.R
import kotlinx.android.synthetic.main.activity_detail.*

/**
 * Created by SKY on 15/11/28 下午2:22.
 */
class DetailActivity : BaseNoPActivity() {
    override fun getLayoutResId(): Int = R.layout.activity_detail

    override fun initialize() {
        toolbarv7.setNavigationIcon(R.mipmap.ic_back)
        setSupportActionBar(toolbarv7)
        toolbarv7.setNavigationOnClickListener { finish() }

        collapsing.title = "我的课程"
    }

    fun checkin(view: View) =
            Snackbar.make(view, "checkin success!", Snackbar.LENGTH_SHORT).show()
}